extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/QuarkusHttpUser.java

```diff
@@ -92,7 +94,10 @@ public static SecurityIdentity getSecurityIdentityBlocking(RoutingContext routin
             return deferred.await().indefinitely();
         }
         if (identityProviderManager != null) {
-            return identityProviderManager.authenticate(AnonymousAuthenticationRequest.INSTANCE).await().indefinitely();
+            return identityProviderManager
+                    .authenticate(setRoutingContextAttribute(new AnonymousAuthenticationRequest(), routingContext))
```

Comment:
Why can't we reuse the same instance representing an anonymous request ?

Reply:
because same instance would be used by multiple HTTP requests, therefore code would access random RoutingContext

Reply:
@michalvavrik Got it, thanks